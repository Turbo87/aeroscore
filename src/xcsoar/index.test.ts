import fs = require('fs');

import {read} from './';

describe('XCSoar - TaskReader - read()', () => {
  it('reads "2017-07-15-lev" task correctly', () => {
    let xml = fs.readFileSync(`${__dirname}/../../fixtures/2017-07-15-lev.tsk`, 'utf8');
    expect(read(xml)).toMatchSnapshot();
  });

  it('reads "2017-07-17-lev" task correctly', () => {
    let xml = fs.readFileSync(`${__dirname}/../../fixtures/2017-07-17-lev.tsk`, 'utf8');
    expect(read(xml)).toMatchSnapshot();
  });

  it('reads "Zwickau" task correctly', () => {
    let xml = fs.readFileSync(`${__dirname}/../../fixtures/Zwickau-2018/C_S/2018-07-30/task.tsk`, 'utf8');
    expect(read(xml)).toMatchSnapshot();
  });
});
