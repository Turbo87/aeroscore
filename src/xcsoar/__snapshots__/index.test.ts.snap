// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XCSoar - TaskReader - read() reads "2017-07-15-lev" task correctly 1`] = `
Object {
  "aat_min_time": 12600,
  "points": Array [
    Object {
      "observation_zone": Object {
        "length": 20000,
        "type": "Line",
      },
      "type": "Start",
      "waypoint": Object {
        "altitude": 86,
        "location": Object {
          "latitude": 51.141383,
          "longitude": 6.9852833,
        },
        "name": "Langenfeld-Wiescheid",
      },
    },
    Object {
      "observation_zone": Object {
        "radius": 10000,
        "type": "Cylinder",
      },
      "type": "Area",
      "waypoint": Object {
        "altitude": 190,
        "location": Object {
          "latitude": 50.824167,
          "longitude": 6.1869500,
        },
        "name": "Aachen Merzbrück",
      },
    },
    Object {
      "observation_zone": Object {
        "radius": 30000,
        "type": "Cylinder",
      },
      "type": "Area",
      "waypoint": Object {
        "altitude": 205,
        "location": Object {
          "latitude": 50.559717,
          "longitude": 7.1375000,
        },
        "name": "Bad Neuenahr",
      },
    },
    Object {
      "observation_zone": Object {
        "radius": 10000,
        "type": "Cylinder",
      },
      "type": "Area",
      "waypoint": Object {
        "altitude": 7,
        "location": Object {
          "latitude": 50.888883,
          "longitude": 6.4916667,
        },
        "name": "058Hambach Süd",
      },
    },
    Object {
      "observation_zone": Object {
        "radius": 10000,
        "type": "Cylinder",
      },
      "type": "Area",
      "waypoint": Object {
        "altitude": 200,
        "location": Object {
          "latitude": 51.176950,
          "longitude": 7.2000000,
        },
        "name": "110Remscheid Bhf",
      },
    },
    Object {
      "observation_zone": Object {
        "radius": 2000,
        "type": "Cylinder",
      },
      "type": "Finish",
      "waypoint": Object {
        "altitude": 130,
        "location": Object {
          "latitude": 51.097783,
          "longitude": 7.0369500,
        },
        "name": "002Zielkreis",
      },
    },
  ],
  "type": "AAT",
}
`;

exports[`XCSoar - TaskReader - read() reads "2017-07-17-lev" task correctly 1`] = `
Object {
  "aat_min_time": 10800,
  "points": Array [
    Object {
      "observation_zone": Object {
        "length": 20000,
        "type": "Line",
      },
      "type": "Start",
      "waypoint": Object {
        "altitude": 50,
        "location": Object {
          "latitude": 51.081950,
          "longitude": 6.9322167,
        },
        "name": "095Monheim AB Dreieck",
      },
    },
    Object {
      "observation_zone": Object {
        "type": "Keyhole",
      },
      "type": "Turn",
      "waypoint": Object {
        "altitude": 100,
        "location": Object {
          "latitude": 50.999167,
          "longitude": 6.2772167,
        },
        "name": "084Linnich Kontrollpunkt",
      },
    },
    Object {
      "observation_zone": Object {
        "type": "Keyhole",
      },
      "type": "Turn",
      "waypoint": Object {
        "altitude": 86,
        "location": Object {
          "latitude": 51.176667,
          "longitude": 6.1922167,
        },
        "name": "113Ritzeroder Duenen",
      },
    },
    Object {
      "observation_zone": Object {
        "type": "Keyhole",
      },
      "type": "Turn",
      "waypoint": Object {
        "altitude": 70,
        "location": Object {
          "latitude": 50.965000,
          "longitude": 6.6077833,
        },
        "name": "021Bergheim",
      },
    },
    Object {
      "observation_zone": Object {
        "radius": 2000,
        "type": "Cylinder",
      },
      "type": "Finish",
      "waypoint": Object {
        "altitude": 130,
        "location": Object {
          "latitude": 51.097783,
          "longitude": 7.0369500,
        },
        "name": "002Zielkreis",
      },
    },
  ],
  "type": "RT",
}
`;

exports[`XCSoar - TaskReader - read() reads "Zwickau" task correctly 1`] = `
Object {
  "aat_min_time": null,
  "points": Array [
    Object {
      "observation_zone": Object {
        "length": 20000,
        "type": "Line",
      },
      "type": "Start",
      "waypoint": Object {
        "altitude": null,
        "location": Object {
          "latitude": 50.682500,
          "longitude": 12.586667,
        },
        "name": "AP Zwickau Ost A72 BAB011",
      },
    },
    Object {
      "observation_zone": Object {
        "type": "Keyhole",
      },
      "type": "Turn",
      "waypoint": Object {
        "altitude": null,
        "location": Object {
          "latitude": 50.763333,
          "longitude": 13.305000,
        },
        "name": "Mittelsaida",
      },
    },
    Object {
      "observation_zone": Object {
        "type": "Keyhole",
      },
      "type": "Turn",
      "waypoint": Object {
        "altitude": null,
        "location": Object {
          "latitude": 50.395833,
          "longitude": 12.343333,
        },
        "name": "Schoeneck Bf",
      },
    },
    Object {
      "observation_zone": Object {
        "type": "Keyhole",
      },
      "type": "Turn",
      "waypoint": Object {
        "altitude": null,
        "location": Object {
          "latitude": 50.355000,
          "longitude": 11.168333,
        },
        "name": "Sonneberg Bf",
      },
    },
    Object {
      "observation_zone": Object {
        "radius": 5000,
        "type": "Cylinder",
      },
      "type": "Finish",
      "waypoint": Object {
        "altitude": null,
        "location": Object {
          "latitude": 50.703333,
          "longitude": 12.459167,
        },
        "name": "Zwickau FP EDBI",
      },
    },
  ],
  "type": "RT",
}
`;
